Core/Rewriters/OrderedMockRewriter.cs

```diff
@@ -0,0 +1,139 @@
+      }
+
+      var rhinoMocksMockRepositoryCompilationSymbol = Model.Compilation.GetTypeByMetadataName ("Rhino.Mocks.MockRepository");
+      if (rhinoMocksMockRepositoryCompilationSymbol == null)
+      {
+        throw new InvalidOperationException ("Rhino.Mocks cannot be found.");
+      }
+
+      var moqSetupSymbol = Model.Compilation.GetTypeByMetadataName ("Moq.Mock`1");
+      var moqCallbackSymbol = Model.Compilation.GetTypeByMetadataName ("Moq.Language.ICallback");
+      var moqReturnsSymbol = Model.Compilation.GetTypeByMetadataName ("Moq.Language.IReturns`2");
+      var moqVerifiableSymbol = Model.Compilation.GetTypeByMetadataName ("Moq.Language.IVerifies");
+      if (moqSetupSymbol == null || moqCallbackSymbol == null || moqReturnsSymbol == null || moqVerifiableSymbol == null)
+      {
+        throw new InvalidOperationException ("Moq cannot be found.");
+      }
+
+      var moqSymbols = GetAllMoqSymbols (moqSetupSymbol, moqCallbackSymbol, moqReturnsSymbol, moqVerifiableSymbol).ToList();
+
+      var usingStatements = GetRhinoMocksOrderedUsingStatements (node, rhinoMocksMockRepositoryCompilationSymbol).ToList();
+
+      var treeWithTrackedNodes = node.TrackNodes (usingStatements);
+      for (var i = 0; i < usingStatements.Count; i++)
+      {
+        var usingStatement = usingStatements[i];
+        var trackedUsingStatement = treeWithTrackedNodes.GetCurrentNode (usingStatement);
+        var statements = ReplaceExpressionStatements (((BlockSyntax) usingStatement.Statement).Statements, moqSymbols, i);
+
+        treeWithTrackedNodes = treeWithTrackedNodes.ReplaceNode (trackedUsingStatement, statements);
+      }
+
+      return treeWithTrackedNodes;
+    }
+
+    private static IEnumerable<ISymbol> GetAllMoqSymbols (
+        INamedTypeSymbol moqSetupSymbol,
+        INamedTypeSymbol moqCallbackSymbol,
+        INamedTypeSymbol moqReturnsSymbol,
+        INamedTypeSymbol moqVerifiableSymbol)
+    {
+      return moqSetupSymbol.GetMembers ("Setup")
+          .Concat (moqCallbackSymbol.GetMembers ("Callback"))
+          .Concat (moqReturnsSymbol.GetMembers ("Returns"))
+          .Concat (moqVerifiableSymbol.GetMembers ("Verifiable"));
+    }
+
+    private IEnumerable<SyntaxNode> ReplaceExpressionStatements (
+        SyntaxList<StatementSyntax> statements,
+        IReadOnlyCollection<ISymbol> moqSymbols,
+        int current)
+    {
+      var nodesToBeReplaced = GetAllMoqExpressionStatements (statements, moqSymbols).ToList();
+      for (var i = 0; i < statements.Count; i++)
+      {
+        var statement = statements[i];
+        if (!nodesToBeReplaced.Any (s => s.IsEquivalentTo (statement, false)))
+        {
+          continue;
+        }
+
+        var firstIdentifierName = statement.GetFirstIdentifierName();
+        var newExpressionStatement = statement.ReplaceNode (
+            firstIdentifierName,
+            MoqSyntaxFactory.InSequenceExpression (firstIdentifierName, (current + 1).ToString()));
```

Comment:
The calling loop should rather start at 1, that way no addition is needed here.

```suggestion
            MoqSyntaxFactory.InSequenceExpression (firstIdentifierName, (current).ToString()));
```